### PR TITLE
[xh]Generate comment for code

### PR DESCRIPTION
**Pull request by matrixstone**
# Summary
Generate comment for each function inside the code.
The newly generated comment will be added as a doc string beneath function definition.
Also it can merge into existing doc string if there is error.

Current issues are (1) the output code is not well formatted (2) # comment misses

# Tests
Command to run the code:
`python mage_ai/ai/generator_cmds.py generate-comment-for-block code_file`

code_file is the file contains the code to generate comment for.

Reviewer:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/ai/llm_pipeline_wizard.py [hunk 1]
```diff
@@ -247,6 +257,42 @@ async def async_generate_pipeline_with_description(self, pipeline_description: s
...
+        prompt_template = PromptTemplate(
+            input_variables=[
+                'block_content',
+            ],
+            template=PROMPT_FOR_FUNCTION_COMMENT,
+        )
+        chain = LLMChain(llm=self.llm, prompt=prompt_template)
+        function_comments = await chain.arun(block_content=block_content)
+        function_comments_json = json.loads(function_comments)
+        return self.__insert_comments_in_functions(block_content, function_comments_json)
```
Comment (dy46) [inline mage_ai/ai/llm_pipeline_wizard.py:294]: i think the variable names here should be flipped. function_comments here is the json string, and function_comments_json is the object

File: mage_ai/ai/llm_pipeline_wizard.py [hunk 2]
```diff
@@ -247,6 +257,42 @@ async def async_generate_pipeline_with_description(self, pipeline_description: s
         await asyncio.gather(*block_tasks)
         return blocks
 
+    def __insert_comments_in_functions(self, code: str, function_comments_json: dict):
```
Comment (dy46) [inline mage_ai/ai/llm_pipeline_wizard.py:260]: let's use `Dict` instead of `dict` here

**Comment by matrixstone** [comment]
Addressed the comments. Thank you for the review David! @dy46

**Review by dy46 — APPROVED** [review]
LGTM. are we going to try to improve the comments later?

**Pull request merged**